pkg/occlient/occlient.go

```diff
@@ -1148,6 +1147,29 @@ func (c *Client) Delete(labels map[string]string) error {
 
 }
 
+// DeleteServiceInstance takes labels as a input and based on it, deletes respective service instance
+func (c *Client) DeleteServiceInstance(labels map[string]string) error {
+	// convert labels to selector
+	selector := util.ConvertLabelsToSelector(labels)
+	log.Debugf("Selectors used for deletion: %s", selector)
+	// Service Instance
+	log.Debug("Deleting Service Instance")
+	// Listing out serviceInstance because `DeleteCollection` method don't work on serviceInstance
+	svcCatList, err := c.serviceCatalogClient.ServiceInstances(c.namespace).List(metav1.ListOptions{LabelSelector: selector})
+	if err != nil {
+		return errors.Wrap(err, "unable to list service instance")
+	}
+	// Iterating over serviceInstance List and deleting one by one
+	for _, svc := range svcCatList.Items {
+		err = c.serviceCatalogClient.ServiceInstances(c.namespace).Delete(svc.Name, &metav1.DeleteOptions{})
+		if err != nil {
+			return errors.Wrap(err, "unable to delete serviceInstace")
+		}
+	}
+	return nil
+}
```

Comment:
The above function seems a bit too clustered. Could you perhaps add some newlines between separate code sections?